compose/cli/main.py

```diff
@@ -437,6 +438,9 @@ def exec_command(self, options):
 
             sys.exit(call_docker(args))
 
+        if tty and not os.isatty(sys.stdin.fileno()):
```

Comment:
ummm, I didn't see that run also had a -T flag, I gave a quick look now and the implementation of the flag looks completely different.
I'll give a deeper look to see if code can be a bit unified to have a more consistent behaviour.